utils/obsproc/NetCDFToIodaConverter.h

```diff
@@ -58,57 +65,108 @@ namespace gdasapp {
+        nobs = iodaVars.location;
+        oops::Log::debug() << "--- iodaVars.location: " << iodaVars.location << std::endl;
+        oops::Log::debug() << "--- iodaVars.obsVal: " << iodaVars.obsVal << std::endl;
+      }
+
+      // Get the total number of obs across pe's
+      comm.allReduce(nobs, nobs, eckit::mpi::sum());
+      oops::Log::debug() << " my rank : " << myrank
+                         << " Num pe's: " << comm.size()
+                         << " nobs: " << nobs << std::endl;
+      gdasapp::IodaVars iodaVarsAll(nobs);
+
+      // Gather obsVal's
+      gatherObs(comm, iodaVars.obsVal, iodaVarsAll.obsVal);
+      gatherObs(comm, iodaVars.obsError, iodaVarsAll.obsError);
+      gatherObs(comm, iodaVars.preQc, iodaVarsAll.preQc);
+
+      oops::Log::debug() << "--- all nobs: " << iodaVarsAll.obsVal.size() << std::endl;
+      oops::Log::debug() << "--- all obsVal: " << iodaVarsAll.obsVal << std::endl;
+      oops::Log::debug() << "--- all obsError: " << iodaVarsAll.obsError << std::endl;
+      oops::Log::debug() << "--- all preQc: " << iodaVarsAll.preQc << std::endl;
+
+      // Create empty group backed by HDF file
+      if (oops::mpi::world().rank() == 0) {
+        ioda::Group group =
+          ioda::Engines::HH::createFile(
+                                        outputFilename_,
+                                        ioda::Engines::BackendCreateModes::Truncate_If_Exists);
+
+        // Update the group with the location dimension
+        ioda::NewDimensionScales_t
+          newDims {ioda::NewDimensionScale<int>("Location", iodaVars.location)};
+        ioda::ObsGroup ogrp = ioda::ObsGroup::generate(group, newDims);
+
+        // Set up the creation parameters
+        ioda::VariableCreationParameters float_params = createVariableParams<float>();
+        ioda::VariableCreationParameters int_params = createVariableParams<int>();
+
+        // Create the IODA variables
+        ioda::Variable adtIodaDatetime =
+          ogrp.vars.createWithScales<float>("Metadata/dateTime",
+                                            {ogrp.vars["Location"]}, float_params);
+        // TODO(All): Decide on what to use for the Epoch date
+        adtIodaDatetime.atts.add<std::string>("units", {"seconds since 9999-04-15T12:00:00Z"}, {1});
+
+        ioda::Variable adtIodaObsVal =
+          ogrp.vars.createWithScales<float>("ObsValue/"+variable_,
+                                            {ogrp.vars["Location"]}, float_params);
+        ioda::Variable adtIodaObsErr =
+          ogrp.vars.createWithScales<float>("ObsError/"+variable_,
+                                            {ogrp.vars["Location"]}, float_params);
+
+        ioda::Variable adtIodaPreQc =
+          ogrp.vars.createWithScales<int>("PreQC/"+variable_,
+                                            {ogrp.vars["Location"]}, int_params);
+
+        // Write adt obs info to group
+        adtIodaObsVal.writeWithEigenRegular(iodaVars.obsVal);
```

Comment:
Did you mean to write the gathered values?

Reply:
oops ... yes!